### DISCUSSION TITLE
NetBox 4.1.1 No Way to Mass Delete Failed Jobs in UI

### DISCUSSION THREAD

**Discussion by mathewbishop** [netbox-community/netbox#17660] [General]
Hello,

I noticed that in the new Background Tasks UI, the ability to mass delete jobs was removed:

<br/>

![image](https://github.com/user-attachments/assets/46a65d44-f42c-4fd3-93e5-be4815c49470)

<br/>

Has anyone found a good way to do this now that this feature is gone? 

I tried getting into Redis directly, but the way the Redis Keys are set up, I was unable to easily identify which keys represented failed jobs specifically.

**Comment by mathewbishop** [comment]
For those interested, I found my way around the Redis keys for jobs and how to find all the jobs (background tasks) with a status of `failed`. 

It looks like each `job` key in Redis has a corresponding `results` key as well. I wrote a python script for cleaning up all the failed jobs out of Redis, which clears this table.

This script finds all the jobs in Redis with a status of `failed` and each job's corresponding `results` key and deletes them out of Redis. I used this to clean up all my failed background tasks.

Use at own risk:

```py
# Clears failed jobs (aka failed Background Tasks in NetBox) from the redis database
import redis

r = redis.Redis()


failed_jobs = [
        key for key in r.keys('rq:job*')
        if r.hget(key, 'status') == b'failed'
]

results = []
for job in failed_jobs:
    job_id = job.decode('utf-8').split(":")[-1]
    matching_results = r.keys(f"rq:results:{job_id}")[0]
    results.append(matching_results)


print(f"Found {len(failed_jobs)} failed jobs")
print(f"Found {len(results)} matching results.")


result_del_count = 0
for idx, result in enumerate(results):
    r.delete(result)
    result_del_count += 1


job_del_count = 0
for idx, job in enumerate(failed_jobs):
    r.delete(job)
    job_del_count += 1


print(f"Deleted {job_del_count} jobs and {result_del_count} matching results")
```

**Comment by cmcknz77** [comment]
Just checked the demo box and there is still no way to delete these in the GUI in NetBox v4.2.4. Was it deliberately removed? Or is it a bug that's never been raised as such?
Thanks @mathewbishop for the workaround. Can you think of a way to export the job data to a file before deletion? I've currently got some ~30k failed jobs sitting in my redis database and I'd like to be able to refer back to the data...

**Reply by mathewbishop** [reply]
Hey, no problem!

I would think there would be a way to dump the redis database to some kind of file, but I don't work with Redis much so I don't know for sure. Redis may have some built in tools for this, but I would have to research it.
